### ISSUE TITLE
BigQuery: Unparsable error when Jinja template value not wrapped in quotes

### ISSUE THREAD

**Issue by tuzonghua** [sqlfluff/sqlfluff#4896]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parser fails on BigQuery templated SQL like `EXTRACT(YEAR FROM some_date_field) = {{ some_date_param }}`. Error is: `Line 4, Position 41: Found unparsable section: ' ='`. Wrapping the Jinja template in quotes fixes the issue but breaks the SQL because of an int to string comparison.

### Expected Behaviour

Run sqlfluff on the SQL code without encountering a parsing error.

### Observed Behaviour

Output when Jinja template is not wrapped in quotes:
```
== [../../test.sql] FAIL
L:   3 | P:  41 |  PRS | Line 3, Position 41: Found unparsable section: ' ='
L:   3 | P:  43 | L001 | Unnecessary trailing whitespace.
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Relevant section of `sql parse`:
```
[L:  3, P:  1]      |            where_clause:
[L:  3, P:  1]      |                keyword:                                      'WHERE'
[L:  3, P:  6]      |                [META] indent:
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                expression:
[L:  3, P:  7]      |                    function:
[L:  3, P:  7]      |                        function_name:
[L:  3, P:  7]      |                            function_name_identifier:         'EXTRACT'
[L:  3, P: 14]      |                        bracketed:
[L:  3, P: 14]      |                            start_bracket:                    '('
[L:  3, P: 15]      |                            [META] indent:
[L:  3, P: 15]      |                            date_part:                        'YEAR'
[L:  3, P: 19]      |                            whitespace:                       ' '
[L:  3, P: 20]      |                            keyword:                          'FROM'
[L:  3, P: 24]      |                            whitespace:                       ' '
[L:  3, P: 25]      |                            expression:
[L:  3, P: 25]      |                                column_reference:
[L:  3, P: 25]      |                                    naked_identifier:         'some_date_field'
[L:  3, P: 40]      |                            [META] dedent:
[L:  3, P: 40]      |                            end_bracket:                      ')'
[L:  3, P: 41]      |                [META] dedent:
[L:  3, P: 41]      |                unparsable:                                   !! Expected: 'Nothing...'
[L:  3, P: 41]      |                    whitespace:                               ' '
[L:  3, P: 42]      |                    raw:                                      '='
```

Output when Jinja template is wrapped in quotes:
```
All Finished 📜 🎉!
```

Relevant section of `sqlfluff parse`:
```
[L:  3, P:  1]      |            where_clause:
[L:  3, P:  1]      |                keyword:                                      'WHERE'
[L:  3, P:  6]      |                [META] indent:
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                expression:
[L:  3, P:  7]      |                    function:
[L:  3, P:  7]      |                        function_name:
[L:  3, P:  7]      |                            function_name_identifier:         'EXTRACT'
[L:  3, P: 14]      |                        bracketed:
[L:  3, P: 14]      |                            start_bracket:                    '('
[L:  3, P: 15]      |                            [META] indent:
[L:  3, P: 15]      |                            date_part:                        'YEAR'
[L:  3, P: 19]      |                            whitespace:                       ' '
[L:  3, P: 20]      |                            keyword:                          'FROM'
[L:  3, P: 24]      |                            whitespace:                       ' '
[L:  3, P: 25]      |                            expression:
[L:  3, P: 25]      |                                column_reference:
[L:  3, P: 25]      |                                    naked_identifier:         'some_date_field'
[L:  3, P: 40]      |                            [META] dedent:
[L:  3, P: 40]      |                            end_bracket:                      ')'
[L:  3, P: 41]      |                    whitespace:                               ' '
[L:  3, P: 42]      |                    comparison_operator:
[L:  3, P: 42]      |                        raw_comparison_operator:              '='
[L:  3, P: 43]      |                    whitespace:                               ' '
[L:  3, P: 44]      |                    quoted_literal:                           '""'
```

### How to reproduce

Test SQL:
```sql
SELECT some_string_field
FROM `some-project.some_dataset.table`
WHERE EXTRACT(YEAR FROM some_date_field) = {{ some_date_param }}
```

Commands run:
```
$ sqlfluff lint test.sql
$ sqlfluff parse test.sql
```

### Dialect

BigQuery

### Version

sqlfluff: 1.4.5
Python: 3.8.16

### Configuration
```
[sqlfluff]
dialect = bigquery
templater = jinja
exclude_rules = L029,L031,L032

[sqlfluff:rules]
max_line_length = 120
tab_space_size = 2
indent_unit = space
capitalization_policy = upper

[sqlfluff:rules:L030]
# Function names
extended_capitalisation_policy = upper

[sqlfluff:templater:jinja:context]
some_date_param = 2023
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tuzonghua** [comment]
Updated the title because I'm also seeing similar parsing errors in the following BigQuery SQL snippets:
```
WHERE some_date <= DATE_ADD("2023-06-01", INTERVAL {{ some_int_param }} MONTH)

SELECT {{some_int_param}} AS months
```

**Comment by tuzonghua** [comment]
This seems to have resolved itself when I wrap the integer parameter in quotes in the config file.
